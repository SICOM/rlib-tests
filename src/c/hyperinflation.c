/*
 *  Copyright (C) 2016 SICOM Systems, INC.
 *
 *  Author: Zoltán Böszörményi <zboszormenyi@sicom.com>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of version 2 of the GNU General Public
 * License as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program; if not, write to the
 * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

#include <config.h>

#include <stdio.h>
#include <rlib.h>

char *salary[13][2] = {
	[0][0] = "date",
	[0][1] = "salary",

	[1][0] = "1945-01-10",
	[1][1] = "10101010101010101010101010101010101010101010101010.1010101010",

	[2][0] = "1945-02-10",
	[2][1] = "01010101010101010101010101010101010101010101010101.0101010101",

	[3][0] = "1945-03-10",
	[3][1] = "10101010101010101010101010101010101010101010101010.1010101010",

	[4][0] = "1945-04-10",
	[4][1] = "01010101010101010101010101010101010101010101010101.0101010101",

	[5][0] = "1945-05-10",
	[5][1] = "10101010101010101010101010101010101010101010101010.1010101010",

	[6][0] = "1945-06-10",
	[6][1] = "01010101010101010101010101010101010101010101010101.0101010101",

	[7][0] = "1945-07-10",
	[7][1] = "10101010101010101010101010101010101010101010101010.1010101010",

	[8][0] = "1945-08-10",
	[8][1] = "01010101010101010101010101010101010101010101010101.0101010101",

	[9][0] = "1945-09-10",
	[9][1] = "10101010101010101010101010101010101010101010101010.1010101010",

	[10][0] = "1945-10-10",
	[10][1] = "01010101010101010101010101010101010101010101010101.0101010101",

	[11][0] = "1945-11-10",
	[11][1] = "10101010101010101010101010101010101010101010101010.1010101010",

	[12][0] = "1945-12-10",
	[12][1] = "01010101010101010101010101010101010101010101010101.0101010101",
};

int main(int argc, char **argv) {
	rlib *r;

	if (argc == 1) {
		printf("usage: %s [ pdf | xml | txt | csv | html ]\n", argv[0]);
		return 1;
	}

	r = rlib_init();
	rlib_add_datasource_array(r, "local_array");
	rlib_add_query_array_as(r, "local_array", salary, 13, 2, "salary");
	rlib_add_report(r, "hyperinflation.xml");
	rlib_set_output_format_from_text(r, argv[1]);
	rlib_set_output_parameter(r, "debugging", "yes");
	rlib_set_output_parameter(r, "only_quote_strings", "yes");
	rlib_set_locale(r, "hu_HU");
	rlib_execute(r);
	rlib_spool(r);
	rlib_free(r);

	return 0;
}
