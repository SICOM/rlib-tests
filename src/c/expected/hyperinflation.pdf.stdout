%PDF-1.3
%·¾­ª
1 0 obj
<<
/Type /Catalog
/Pages 2 0 R
>>
endobj
2 0 obj
<<
/Type /Pages
/Kids [ 4 0 R ]
/Count 1
>>
endobj
3 0 obj
<<
/Producer (Haru Free PDF Library 2.2.1)
/Title (RLIB Report)
>>
endobj
4 0 obj
<<
/Type /Page
/MediaBox [ 0 0 612 792 ]
/Contents 5 0 R
/Resources <<
/ProcSet [ /PDF /Text /ImageB /ImageC /ImageI ]
/Font <<
/F1 7 0 R
/F2 8 0 R
>>
>>
/Parent 2 0 R
/Rotate 270
>>
endobj
5 0 obj
<<
/Length 6 0 R
>>
stream
1 0 0 1 0 792 cm
0 -1 1 0 0 0 cm
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 11 Tf
/F2 11 Tf
BT
1 0 0 1 14.4 589.23999 Tm
(Hyperinflation salaries in Hungary, 1945) Tj
ET
/F1 11 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 570.76001 Tm
(Total:    ) Tj
ET
BT
1 0 0 1 14.4 561.76001 Tm
(          ) Tj
ET
BT
1 0 0 1 14.4 552.76001 Tm
(          ) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 543.76001 Tm
(Date      Salary in plain number                                                          ) Tj
ET
BT
1 0 0 1 14.4 534.76001 Tm
(          Salary in national currency                                                     ) Tj
ET
BT
1 0 0 1 14.4 525.76001 Tm
(          Salary in international currency                                                ) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 516.76001 Tm
(1945-01-10                                              $10101010101010101010101010101010101010101010101010.10) Tj
ET
BT
1 0 0 1 14.4 507.76001 Tm
(                         *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10 Ft) Tj
ET
BT
1 0 0 1 14.4 498.76001 Tm
(                       HUF  *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 489.76001 Tm
(1945-02-10                                               $1010101010101010101010101010101010101010101010101.01) Tj
ET
BT
1 0 0 1 14.4 480.76001 Tm
(                         **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01 Ft) Tj
ET
BT
1 0 0 1 14.4 471.76001 Tm
(                       HUF  **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 462.76001 Tm
(1945-03-10                                              $10101010101010101010101010101010101010101010101010.10) Tj
ET
BT
1 0 0 1 14.4 453.76001 Tm
(                         *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10 Ft) Tj
ET
BT
1 0 0 1 14.4 444.76001 Tm
(                       HUF  *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 435.76001 Tm
(1945-04-10                                               $1010101010101010101010101010101010101010101010101.01) Tj
ET
BT
1 0 0 1 14.4 426.76001 Tm
(                         **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01 Ft) Tj
ET
BT
1 0 0 1 14.4 417.76001 Tm
(                       HUF  **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 408.76001 Tm
(1945-05-10                                              $10101010101010101010101010101010101010101010101010.10) Tj
ET
BT
1 0 0 1 14.4 399.76001 Tm
(                         *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10 Ft) Tj
ET
BT
1 0 0 1 14.4 390.76001 Tm
(                       HUF  *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 381.76001 Tm
(1945-06-10                                               $1010101010101010101010101010101010101010101010101.01) Tj
ET
BT
1 0 0 1 14.4 372.76001 Tm
(                         **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01 Ft) Tj
ET
BT
1 0 0 1 14.4 363.76001 Tm
(                       HUF  **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 354.76001 Tm
(1945-07-10                                              $10101010101010101010101010101010101010101010101010.10) Tj
ET
BT
1 0 0 1 14.4 345.76001 Tm
(                         *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10 Ft) Tj
ET
BT
1 0 0 1 14.4 336.76001 Tm
(                       HUF  *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 327.76001 Tm
(1945-08-10                                               $1010101010101010101010101010101010101010101010101.01) Tj
ET
BT
1 0 0 1 14.4 318.76001 Tm
(                         **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01 Ft) Tj
ET
BT
1 0 0 1 14.4 309.76001 Tm
(                       HUF  **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 300.76001 Tm
(1945-09-10                                              $10101010101010101010101010101010101010101010101010.10) Tj
ET
BT
1 0 0 1 14.4 291.76001 Tm
(                         *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10 Ft) Tj
ET
BT
1 0 0 1 14.4 282.76001 Tm
(                       HUF  *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 273.76001 Tm
(1945-10-10                                               $1010101010101010101010101010101010101010101010101.01) Tj
ET
BT
1 0 0 1 14.4 264.76001 Tm
(                         **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01 Ft) Tj
ET
BT
1 0 0 1 14.4 255.75999 Tm
(                       HUF  **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 246.75999 Tm
(1945-11-10                                              $10101010101010101010101010101010101010101010101010.10) Tj
ET
BT
1 0 0 1 14.4 237.75999 Tm
(                         *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10 Ft) Tj
ET
BT
1 0 0 1 14.4 228.75999 Tm
(                       HUF  *************10.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010,10) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 219.75999 Tm
(1945-12-10                                               $1010101010101010101010101010101010101010101010101.01) Tj
ET
BT
1 0 0 1 14.4 210.75999 Tm
(                         **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01 Ft) Tj
ET
BT
1 0 0 1 14.4 201.75999 Tm
(                       HUF  **************1.010.101.010.101.010.101.010.101.010.101.010.101.010.101.010.101,01) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 192.75999 Tm
(Total:    ) Tj
ET
BT
1 0 0 1 14.4 183.75999 Tm
(          ) Tj
ET
BT
1 0 0 1 14.4 174.75999 Tm
(          ) Tj
ET
/F1 9 Tf
/F1 9 Tf
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 68.4 174.75999 Tm
(             HUF  *************66.666.666.666.666.666.666.666.666.666.666.666.666.666.666.666.666,67) Tj
ET
/F1 9 Tf
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 68.4 183.75999 Tm
(               *************66.666.666.666.666.666.666.666.666.666.666.666.666.666.666.666.666,67 Ft) Tj
ET
/F1 9 Tf
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 68.4 192.75999 Tm
(                                              $66666666666666666666666666666666666666666666666666.67) Tj
ET
/F1 9 Tf
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 68.4 552.76001 Tm
(             HUF  *************66.666.666.666.666.666.666.666.666.666.666.666.666.666.666.666.666,67) Tj
ET
/F1 9 Tf
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 68.4 561.76001 Tm
(               *************66.666.666.666.666.666.666.666.666.666.666.666.666.666.666.666.666,67 Ft) Tj
ET
/F1 9 Tf
0 0 0 rg
0 0 0 RG
BT
1 0 0 1 68.4 570.76001 Tm
(                                              $66666666666666666666666666666666666666666666666666.67) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf

endstream
endobj
6 0 obj
7914
endobj
7 0 obj
<<
/Type /Font
/BaseFont /Courier
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
8 0 obj
<<
/Type /Font
/BaseFont /Courier-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
xref
0 9
0000000000 65535 f
0000000015 00000 n
0000000064 00000 n
0000000123 00000 n
0000000205 00000 n
0000000403 00000 n
0000008371 00000 n
0000008391 00000 n
0000008486 00000 n
trailer
<<
/Root 1 0 R
/Info 3 0 R
/Size 9
>>
startxref
8586
%%EOF
